go-controller/pkg/config/cni.go

```diff
@@ -71,29 +73,89 @@ func WriteCNIConfig() error {
 
 // ParseNetConf parses config in NAD spec
 func ParseNetConf(bytes []byte) (*ovncnitypes.NetConf, error) {
+	netconf, isRecoverable, err := parseNetConfList(bytes)
```

Comment:
nit: maybe rename from `isRecoverable` to `failedToParseConfListFormat` or something ? 

I think that makes it clearer below, when we attempt to `parseNetConfSingle` - i.e.

```golang
if failedToParseConfListFormat {
			netconf, err = parseNetConfSingle(bytes)
			if err != nil {
				return nil, err
			}
		} else {
			return nil, err
		}
```

just an opinion.